doc/source/user/problem_method.rst

```diff
@@ -139,6 +139,340 @@ fields
+unsplit Godunov method described by
+`Stone & Gardiner (2009) <http://adsabs.harvard.edu/abs/2009NewA...14..139S>`_
+.
+
+The algorithm combines attributes similar to the MUSCL-Hancock with
+the constrained transport method.  In short, the method includes two
+main steps. First the values are integrated to the half
+time-step. Then, the values at the half time-step are used to
+caluclate the change in the quantities over the full time-step. The
+primary representation of the magnetic fields are 3 face-centered
+fields (one for each component), which lie on the cell face matching
+the dimension of the component (e.g. the x component lies on the
+x-face). The method also tracks cell-centered components of the bfield
+which are just the averages of the face-centered components. The
+implementation has been modified to perform reconstruction using
+internal energy inplace of pressure (in typical Enzo fashion).
+
+This method can effectively function as a hydrodynamics method if
+the magnetic fields are all initialized to zero (if they start at
+zero, they will never be modified to non-zero values).
+
+Note that the courant factor should be less than 0.5.
+
+This method only support 3 dimensions.
+
+
+parameters
+----------
+
+.. list-table:: Method ``mhd_vlct`` parameters
+   :widths: 10 5 1 30
+   :header-rows: 1
+   
+   * - Parameter
+     - Type
+     - Default
+     - Description
+   * - ``"density_floor"``
+     - `float`
+     - `none`
+     - `Lower limit on density (must exceed 0)`
+   * - ``"pressure_floor"``
+     - `logical`
+     - `none`
+     - `Lower limit on thermal pressure (must exceed 0)`
+   * - ``"riemann_solver"``
+     - `string`
+     - `hlld`
+     - `name of the Riemann Solver to use`
+   * - ``"half_dt_reconstruct_method"``
+     - `string`
+     - `nn`
+     - `Reconstruction method for half timestep`
+   * - ``"full_dt_reconstruct_method"``
+     - `string`
+     - `plm`
+     - `Reconstruction method for full timestep`
+   * - ``"theta_limiter"``
+     - `float`
+     - `1.5`
+     - `controls dissipation of the "plm"/"plm_enzo" reconstruction
+       method.`
+   * - ``"dual_energy"``
+     - `logical`
+     - `false`
+     - `Whether to use dual-energy formalism`
+   * - ``"dual_energy_eta"``
+     - `float`
+     - `0.001`
+     - `Dual energy parameter eta`
+
+
+fields
+------
+
+.. list-table:: Method ``mhd_vlct`` fields
+   :widths: 5 5 1 30
+   :header-rows: 1
+
+   * - Field
+     - Type
+     - Read/Write
+     - Description   
+   * - ``"density"``
+     - ``enzo_float``
+     - [rw]
+     -    
+   * - ``"velocity_x"``
+     - ``enzo_float``
+     - [rw]
+     -
+   * - ``"velocity_y"``
+     - ``enzo_float``
+     - [rw]
+     -
+   * - ``"velocity_z"``
+     - ``enzo_float``
+     - [rw]
+     -
+   * - ``"total_energy"``
+     - ``enzo_float``
+     - [rw]
+     - specific total energy
+   * - ``"bfield_x"``
+     - ``enzo_float``
+     - [rw]
```

Comment:
I see where you are coming from with this comment and I agree with it (in principle).

However, as things stand the user is responsible for making sure the cell-centered **B**-field is properly initialized. If they don't, the program won't work properly.

I do think that it would probably be good to remove this point of failure. Maybe in the future, we could try to make sure the cell-centered bfield is always computed at the end of the initialization, before the first execution of `EnzoMethodMHD::compute`. But I think that may be beyond the scope of this PR. (I'm happy to open an issue on this topic so that we don't forget)